### PR TITLE
fix: public caret api annotations

### PR DESCRIPTION
**Pull request by taro-yu**
## Description

The purpose of this PR is to modify the annotations of functions published in the API list (https://tier4.github.io/caret_analyze/latest/). In this PR, annotations on docs and annotation corrections on code were made. Since it was necessary to modify the code for some of the processes, this was done in the separate PR (https://github.com/tier4/caret_analyze/pull/490).
## Related links

- https://tier4.atlassian.net/browse/RT2-1277

- https://tier4.atlassian.net/wiki/spaces/EMBIV/pages/3008168691/A+API

## Pre-review checklist for the PR author

- [x] I've confirmed the [contribution guidelines](https://github.com/tier4/caret/blob/main/.github/CONTRIBUTING.md).
- [x] The PR is ready for review.

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR has been properly tested.
- [ ] The PR has been reviewed.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] (Optional) The PR has been properly tested with CARET_report verification.
- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

### CODE REVIEW

**Review by rokamu623 — CHANGES_REQUESTED** [review]
Unify the annotations for variable length arguments. (`*args: type` or `args: tuple[type]`?)

File: src/caret_analyze/runtime/application.py [hunk 1]
```diff
@@ -449,13 +449,13 @@ def get_node_path(
...
             topic name which the node subscribes.
-        publish_topic_name : str | None
+        publish_topic_name : str
             topic name which the node publishes.
-        subscription_construction_order : int | None
+        subscription_construction_order : int
             A construction order of subscription.
-        publisher_construction_order : int | None
+        publisher_construction_order : int
             A construction order of publisher.
```
Comment (rokamu623) [inline src/caret_analyze/runtime/application.py:459]: Changes to the source code should be made in a separate PR.

**Review by rokamu623 — CHANGES_REQUESTED** [review]

File: src/caret_analyze/runtime/application.py (hunk 1 above)
Comment (taro-yu) [reply src/caret_analyze/runtime/application.py]: Supported by the PR (https://github.com/tier4/caret_analyze/pull/490). Please review.

**Comment by taro-yu** [comment]
> Unify the annotations for variable length arguments. (`*args: type` or `args: tuple[type]`?)

Fixed in https://github.com/tier4/caret_analyze/pull/488/commits/24381145b61689bd406f643225607b319c11de7e and https://github.com/tier4/caret_analyze/pull/488/commits/d5a3e263ec8408f8aa24301fb5ed0b6b119de116.

**Review by rokamu623 — APPROVED** [review]
LGTM!

**Review by rokamu623 — APPROVED** [review]
`caret_report` verification is succeed!
image
